runtime/vm/vmthread.cpp

```diff
@@ -260,6 +260,13 @@ allocateVMThread(J9JavaVM * vm, omrthread_t osThread, UDATA privateFlags, void *
 	newThread->ffiArgCount = 0;
 #endif /* JAVA_SPEC_VERSION >= 16 */
 
+#if JAVA_SPEC_VERSION >= 19
+	newThread->currentContinuation = NULL;
+	newThread->pinnedStateCounter = 0;
+	newThread->carrierThreadObject = threadObject;
```

Comment:
Please change `thread.cpp:Java_java_lang_Thread_currentCarrierThread()` accordingly, and remove the `ToDo` comment.